python/tvm/target/detect_target.py

```diff
@@ -58,6 +58,17 @@ def _detect_rocm(dev: Device) -> Target:
     )
 
 
+def _detect_opencl(dev: Device) -> Target:
+    return Target(
+        {
+            "kind": "opencl",
```

Comment:
In some cases, we can initialize OpenCL target with specifying specific device, e.g.:
```
target = Target("opencl -device=mali", host="llvm -mtriple=aarch64-linux-gnu")
or
target = Target("opencl -device=adreno", host="llvm -mtriple=arm64-linux-android")
```
Can we handle these cases somehow?

Reply:
Yes, I have verified on Orange Pi.

![Screenshot from 2024-04-09 12-56-06](https://github.com/apache/tvm/assets/164936748/1c25fe7d-f04d-4211-bc81-1d14cf15cc12)

Reply:
Sorry, probably my question wasn't clear. Yes, I know that we can get such information from target. But I'm curious if we can get this information from the device somehow? 

Probably, we can specify by an addition parameter to `Target.from_device(input_device)`. For example: `Target.from_device(input_device, '-device=adreno')` will specify target as `"kind": "opencl -device=adreno"`. Does it make sense?

Reply:
It seems there is no argument for '-device=adreno' in API [Target.from_device.](https://github.com/apache/tvm/blob/main/python/tvm/target/target.py#L154)

This PR intends to fix weight conversion for MLC LLM on OrangePi ([#1557](https://github.com/mlc-ai/mlc-llm/issues/1557)), assuming there are no other issues. I would like to merge it first. If you have any suggestions, perhaps we can create another issue to discuss. Thanks.